### PR TITLE
Add Algolia's docSearch

### PR DESCRIPTION
**Pull request by LukyVj**
Hi, so here it is! 
A [docSearch](https://community.algolia.com/docsearch/) implementation on Styleguidist website! 

Desktop: 

| Closed  | Open |
| ------------- | ------------- |
|   |   |

Mobile: 

| Closed  | Open | Searched  |
| ------------- | ------------- | ------------- |
|   |   |   |

### CODE REVIEW

File: templates/components/Footer.jsx [hunk 1]
```diff
@@ -2,10 +2,42 @@ import Container from 'tamia/lib/components/Container';
...
+		    src="//cdn.jsdelivr.net/docsearch.js/2/docsearch.min.js" onload={`
+		        javascript:docsearch({
+		          apiKey: '0bd0dc976499f3a333c9d26416b4fee1',
+							indexName: 'react_styleguidist',
+							inputSelector: '#searchbox',
+		          debug: false
+		        })
+		    `}
+		></script>
+			<svg xmlns="http://www.w3.org/2000/svg">
```
Comment (sapegin) [inline templates/components/Footer.jsx]: Looks like you’re not using this SVG anymore.

File: templates/components/SearchBox.jsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import s from './SearchBox.pcss';
+
+export default function Search() {
+	return (
+		<div class={s.searchboxHolder}>
+			<svg id="search-icon" width="40" height="40" viewBox="0 0 40 40" onclick="this.parentNode.childNodes[1].childNodes[0].focus()" class={s.searchboxIcon}><path d="M26.806 29.012a16.312 16.312 0 0 1-10.427 3.746C7.33 32.758 0 25.425 0 16.378 0 7.334 7.333 0 16.38 0c9.045 0 16.378 7.333 16.378 16.38 0 3.96-1.406 7.593-3.746 10.426L39.547 37.34c.607.608.61 1.59-.004 2.203a1.56 1.56 0 0 1-2.202.004L26.808 29.012zm-10.427.627c7.32 0 13.26-5.94 13.26-13.26 0-7.325-5.94-13.26-13.26-13.26-7.325 0-13.26 5.935-13.26 13.26 0 7.32 5.935 13.26 13.26 13.26z" fill-rule="evenodd" class={s.searchboxIconPath}/></svg>
```
Comment (sapegin) [inline templates/components/SearchBox.jsx]: Do we still need this ID?

**Review by sapegin — COMMENTED** [review]
THIS IS JUST AMAZING!! 🍕 🚀 🐙

File: templates/components/Footer.jsx (hunk 1 above)
Comment (LukyVj) [reply templates/components/Footer.jsx]: You're right! My bad!

File: templates/components/SearchBox.jsx (hunk 2 above)
Comment (LukyVj) [reply templates/components/SearchBox.jsx]: Nope! My bad!

**Comment by LukyVj** [comment]
Hi, do you need anything else before merging ?

**Comment by sapegin** [comment]
Could you remove that unused SVG and id attribute? :-)

**Comment by LukyVj** [comment]
🚀

**Comment by sapegin** [comment]
Looks cool! https://593e490ecf321c70dd7ffa13--styleguidist.netlify.com/

One strange thing: scroll to “Interactive playground”, wait a few seconds — animation is not aligned and it works fine on https://react-styleguidist.js.org/ — not sure if it was caused by the changes in this PR ;-|

**Comment by LukyVj** [comment]
Yeah I see it, I'm not sure why it happens.

**Comment by LukyVj** [comment]
Yes so apparently it's because of the css for docsearch, let me investigate

**Comment by LukyVj** [comment]
Ok so weirdly enough, on local, I don't see this bug. 
May need extra css to fix it. 

Let me try something first.

**Comment by LukyVj** [comment]
@sapegin Can you share with me the new netlify url please! :)

**Comment by LukyVj** [comment]
Okay, I got it :)

These are the bad boys : 
```
@-webkit-keyframes a {
    0% {
        -webkit-transform: translate3d(-20%,0,0);
        transform: translate3d(-20%,0,0);
        opacity: 0
    }

    to {
        -webkit-transform: none;
        transform: none;
        opacity: 1
    }
}

@keyframes a {
    0% {
        -webkit-transform: translate3d(-20%,0,0);
        transform: translate3d(-20%,0,0);
        opacity: 0
    }

    to {
        -webkit-transform: none;
        transform: none;
        opacity: 1
    }
}
```


They are added by DocSearch, and your animation got the same name: `a`

**Comment by sapegin** [comment]
Is it a minified name?

**Comment by LukyVj** [comment]
But it's weird, seems like the compile code animation name is `a`, but when I look at the sources, I see the animation name is "Image" 

On netlify : https://puu.sh/whNNF/f3cab2386f.png
On local : https://puu.sh/whNP9/c83188b747.png

**Comment by LukyVj** [comment]
I guess it is yes.

**Comment by sapegin** [comment]
Yes, I think it’s some kind of minification that makes the same name for both CSS files ;-|

**Comment by LukyVj** [comment]
On the CDN where we host the code for docsearch, the animation name is "a" by default. https://puu.sh/whNS6/8eca0cdf62.png

https://cdn.jsdelivr.net/docsearch.js/2/docsearch.min.css

You will have to remove the minification of the animation on your side! 

But I'm not quite sure how to do that.. :s

**Comment by LukyVj** [comment]
What we can do, is to remove the `<link [...]>` to the docsearch, and load it directly in your css, with a different animation name.

**Comment by sapegin** [comment]
Looks like we need to define it like this `@keyframes :global(xxx)`. Could you please try it in your branch? I think it’s because I don’t use hashes in CSS Module class names (because I don’t want to rebuild HTML after every CSS change).

**Comment by sapegin** [comment]
I also think you may want to use more unique name to prevent such conflicts in the future ;-)

**Comment by LukyVj** [comment]
I have a fix already

**Comment by LukyVj** [comment]
let me push it

**Comment by LukyVj** [comment]
You're good to go @sapegin

**Comment by LukyVj** [comment]
Well, I don't get what happen, this is what I have on local : https://puu.sh/whOEZ/f08094c6ce.png

And this is what I have on netlify : https://puu.sh/whOGj/b99408246c.png

**Comment by LukyVj** [comment]
Please let me fixing it and cleaning the code before merging :)

**Comment by LukyVj** [comment]
Okay so, can you try on your end? I'm not sure why the file `styles/docsearch.pcss` seems to be ignored.. :/

**Comment by LukyVj** [comment]
Ok so, I'm sorry, I'm not good enough at tamia to be able to fix it. 

The simplet solution would be to keep using docsearch.css served by CDN, and prevent your build system to rename your animation.

**Comment by sapegin** [comment]
OK, I’ll try it, thanks!

**Comment by sapegin** [comment]
I’ve disabled CSS name mangling in tamia-build so it should work now.

**Comment by sapegin** [comment]
Looks like it finally works! Thank you very much!! I’ll merge and clean up a bit later 🦄 🍕

**Pull request merged**